tests/helpers/socket.py

```diff
@@ -57,7 +57,10 @@ def connect():
 
 def close(sock):
     """Shutdown and close the socket."""
-    sock.shutdown(socket.SHUT_RDWR)
+    try:
+        sock.shutdown(socket.SHUT_RDWR)
+    except Exception:
```

Comment:
It's basically impossible to run most tests that use live sockets on Win32 without this. I think it might be because shutting down a closed socket on other platforms isn't considered an error?